pkg/providers/equinix.go

```diff
@@ -45,6 +49,21 @@ func EquinixProviderFactory(providerInfo string, secretData map[string][]byte, l
 		}
 	}
 
+	if config.UserData != "" {
+		// Provided userdata from secret should be base64 encoded
+		decodedBytes, err := base64.StdEncoding.DecodeString(config.UserData)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding userdata: %w", err)
+		}
+
+		stringScript := string(decodedBytes)
+		if err := utils.ValidateCloudInitScript(stringScript); err != nil {
```

Comment:
Since the provider it's potentially invoked for every reconcile step, we do want to keep it as much as possible light and non-blocking, avoid as much as possible lengthy operations.
It's currently responsibility of the user writing the provider configuration to ensure it's correct; it could probably make sense to have a separate validation step in the cipool controller (whenever a secret gets updated), but we never implemented it as too much expensive for a usually one-off operation

Reply:
Ack. Fare point. I think it could be beneficial to avoid the lease attempt in case of wrong script, but it not that important. I can add it to the CIPool controller or drop it. What do you prefer ? @andfasano

Reply:
My preference would be to drop it until we'll get more cases

Reply:
Done

Reply:
Do you want to drop the base64 conversion as well ?